### PR TITLE
Upgrade to dune 2.0

### PR DESCRIPTION
**Pull request by reynir**
I found that dune would attempt to compile ocaml-dns to javascript if its source was included as a subdirectory in happy-eyeballs. With help from people on the discord I had two solutions, and one of them was to upgrade to dune-lang 2.0.

### CODE REVIEW

**Comment by hannesm** [comment]
ok, but why then `2.0.1`? to me, `2.0` or `2.0.0` would be more intuitive (dune 2.0.0 is also marked as unavailable).

**Comment by reynir** [comment]
Sure. I used 2.0.1 as 2.0.0 was unavailable. I'll change it.

File: dune-project [hunk 1]
```diff
@@ -1,2 +1,2 @@
-(lang dune 1.2)
+(lang dune 2.0)
 (name dns)
```
Comment (hannesm) [inline dune-project:2]:
```suggestion
(name dns)
(formatting disabled)
```

Dune 2.0 imposes some ocamlformat stuff, which we're not using here (yet?).

Comment (reynir) [reply dune-project:2]: Ah, I wasn't aware. Maybe we should?

**Comment by hannesm** [comment]
thanks

**Pull request merged**